lib/storage/bucket.js

```diff
@@ -198,6 +198,10 @@ Bucket.prototype.delete = function(callback) {
  * var file = bucket.file('my-existing-file.png');
  */
 Bucket.prototype.file = function(name) {
+  if (!name) {
```

Comment:
Sorry for the confusion, just returned from 3-day conference and catching up. I wanted to make sure that @coderaiser knows that this PR doesn't really do anything for his original issue and just now catches the original error one function call sooner. It's helpful to have that test in place because I think it was missed. I'm fine with merging this as long as its understood that File is NEVER to be used independently of Bucket.

As an aside, maybe naming the File filename something like `bucket.file.js` to show the hard hierarchical relationship between `bucket.js` and `file.js`.